lib/ansible/modules/windows/win_data_deduplication.ps1

```diff
@@ -0,0 +1,174 @@
+      }
+    } else {
+      Fail-Json $result "Unsupported state option, it can only be 'enabled' or 'disabled'."
+    }
+
+    $result.msg += " Deduplication on volume: "+$volume.DriveLetter+" set to: $state"
+    $result.changed = $true
+  }
+
+  if ($state -eq 'enabled') {
+    if ($null -ne $settings) {
+      Set-DataDedupJobSettings -volume $volume -settings $settings
+    }
+
+    if ($null -ne $dedup_job) {
+      Start-DataDedupJob -volume $volume -dedup_job $dedup_job
+    }
+  }
+}
+
+function Set-DataDedupJobSettings ($volume, $settings) {
+
+  try {
+    $dedup_info = Get-DedupVolume -Volume "$($volume.DriveLetter):"
+  } catch {
+    $dedup_info = $null
+  }
+
+  ForEach ($key in $settings.keys) {
+    $update_key = $key
+    $update_value = $settings.$($key)
+
+    if ($update_key -eq "MinimumFileSize" -and $update_value -lt 32768) {
+      $update_value = 32768
+    }
+
+    try {
+      $current_value = ($dedup_info | Select-Object -ExpandProperty $update_key)
+    } catch {
+      $current_value = $null
+    }
+
+    if ($update_value -ne $current_value) {
+      $command_param = @{
+        $($update_key) = $update_value
+      }
+
+      # Set-DedupVolume -Volume <String>`
+      #                 -NoCompress <bool> `
+      #                 -MinimumFileAgeDays <UInt32> `
+      #                 -MinimumFileSize <UInt32> (minimum 32768)
+      try {
+        Set-DedupVolume -Volume "$($volume.DriveLetter):" @command_param
+        $result.msg += " Setting DedupVolume settings for $update_key"
+
+      } catch {
+        Fail-Json $result $_.Exception.Message
+      }
+
+      $result.changed = $true
+    }
+  }
+
+}
+
+function Start-DataDedupJob($volume, $dedup_job) {
+
+  $dedup_job_queue = Get-DedupJob -Volume "$($volume.DriveLetter):" -ErrorAction SilentlyContinue
+
+  if(( $dedup_job_queue | Select-Object -ExpandProperty State) -ne "Queued" ) {
+
+    $command_param = @{}
+
+    ForEach ($key in $dedup_job.keys) {
+
+      $update_key = $key
+      $update_value = $dedup_job.$($key)
+
+      $command_param.Add($($update_key), $update_value)
+
+    }
+
+    # Start-DedupJob -Volume <String> `
+    #          -Type <String>
+    $dedup_job_start = Start-DedupJob -Volume "$($volume.DriveLetter):" @command_param
+    $result.msg += " DedupJob "+($dedup_job_start | Select-Object -ExpandProperty State)
+
+    $result.changed = $true
+  } else {
+    $result.msg += " A DedupJob is already queued for that volume."
+  }
+
+}
+
+
+# Install required feature
+$feature_name = "FS-Data-Deduplication"
+$feature = Install-WindowsFeature -Name $feature_name
+
+if ($feature.RestartNeeded -eq 'Yes') {
+  $result.reboot_required = $true
+  Fail-Json $result "$feature_name was installed but requires Windows to be rebooted to work."
+  Exit-Json -obj $result
+}
+
+try {
+  $volume = Get-Volume -DriveLetter $drive_letter
+} catch {
+  Fail-Json $result $_.Exception.Message
+}
+if ($null -ne $volume) {
```

Comment:
I don't see a scenario where this would be `$null` as we set `$ErrorActionPreference = "Stop"` when running modules. In that case `Get-Volume` will fail on a missing volume and not reach this code.